avm/ptn/app-service-lza/hosting-environment/modules/supporting-services/modules/key-vault.bicep

```diff
@@ -0,0 +1,137 @@
+param spokePrivateEndpointSubnetName string
+
+@description('Optional. The name of the private endpoint to be created for Key Vault. If left empty, it defaults to "<resourceName>-pep')
+param keyVaultPrivateEndpointName string = 'keyvault-pep'
+
+@description('Optional. Diagnostic Settings for the Key Vault.')
+param diagnosticSettings diagnosticSettingFullType[]
+
+param appServiceManagedIdentityPrincipalId string
+
+// ------------------
+// VARIABLES
+// ------------------
+
+var vaultDnsZoneName = 'privatelink.vaultcore.azure.net'
+var spokeVNetIdTokens = split(spokeVNetResourceId, '/')
+var spokeSubscriptionId = spokeVNetIdTokens[2]
+var spokeResourceGroupName = spokeVNetIdTokens[4]
+var spokeVNetName = spokeVNetIdTokens[8]
+var virtualNetworkLinks = concat(
+  [
+    {
+      virtualNetworkResourceId: spokeVNetResourceId
+      registrationEnabled: false
+    }
+  ],
+  (!empty(hubVNetResourceId))
+    ? [
+        {
+          virtualNetworkResourceId: hubVNetResourceId
+          registrationEnabled: false
+        }
+      ]
+    : []
+)
+// ------------------
+// RESOURCES
+// ------------------
+
+resource vnetSpoke 'Microsoft.Network/virtualNetworks@2022-01-01' existing = {
+  scope: resourceGroup(spokeSubscriptionId, spokeResourceGroupName)
+  name: spokeVNetName
+}
+
+resource spokePrivateEndpointSubnet 'Microsoft.Network/virtualNetworks/subnets@2022-07-01' existing = {
+  parent: vnetSpoke
+  name: spokePrivateEndpointSubnetName
+}
+
+module vaultdnszone 'br/public:avm/res/network/private-dns-zone:0.7.0' = {
+  name: 'keyvaultDnsZoneDeployment-${uniqueString(resourceGroup().id)}'
+  params: {
+    name: vaultDnsZoneName
+    location: 'global'
+    tags: tags
+    enableTelemetry: enableTelemetry
+    virtualNetworkLinks: virtualNetworkLinks
+  }
+}
+
+module keyvault 'br/public:avm/res/key-vault/vault:0.12.1' = {
+  name: 'vault-${uniqueString(resourceGroup().id)}'
+  params: {
+    name: keyVaultName
+    location: location
+    tags: tags
+    enableTelemetry: enableTelemetry
+    sku: 'standard'
+    networkAcls: {
+      bypass: 'AzureServices'
+      defaultAction: 'Deny'
+    }
+    enableSoftDelete: true
+    softDeleteRetentionInDays: 7
+    enablePurgeProtection: null
+    publicNetworkAccess: 'Disabled'
+    enableRbacAuthorization: true
+    enableVaultForDeployment: true
+    privateEndpoints: [
+      {
+        name: keyVaultPrivateEndpointName
+        privateDnsZoneGroup: {
+          privateDnsZoneGroupConfigs: [
+            {
+              privateDnsZoneResourceId: vaultdnszone.outputs.resourceId
+            }
+          ]
+        }
+        subnetResourceId: spokePrivateEndpointSubnet.id
+      }
+    ]
+    diagnosticSettings: diagnosticSettings
+    roleAssignments: [
+      {
+        principalId: appServiceManagedIdentityPrincipalId
+        roleDefinitionIdOrName: '4633458b-17de-408a-b874-0445c86b69e6' //Key Vault Secrets User
```

Comment:
Note, you can actually use the name here as its tracked in the  key vault file

https://github.com/Azure/bicep-registry-modules/blob/ff8dc79cd4cc2794692fee8d6ec7b4e51b36e572/avm/res/key-vault/vault/main.bicep#L131-L134

So you can write `roleDefinitionIdOrName: 'Key Vault Secrets User'`